plugins/mapper-murmur3/src/main/java/org/elasticsearch/index/mapper/murmur3/Murmur3FieldMapper.java

```diff
@@ -111,6 +111,19 @@ public String typeName() {
             return new SortedNumericIndexFieldData.Builder(name(), NumericType.LONG);
         }
 
+        @Override
+        public ValueFetcher valueFetcher(MapperService mapperService, SearchLookup searchLookup, String format) {
+            if (format != null) {
+                throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] doesn't support formats.");
+            }
+            return new SourceValueFetcher(name(), mapperService, false) {
+                @Override
+                protected String parseSourceValue(Object value) {
+                    return value.toString();
+                }
+            };
+        }
```

Comment:
Yes, this can definitely be replaced with a common implementation in a lot of cases.  I didn't want to do that in this case because the PR is big enough as it is...

Reply:
makes sense

Reply:
+1 we could certainly add some shared `SourceValueFetcher` implementations, for example an 'identity' fetcher and a 'string' fetcher. When introducing the feature I just tried to keep the design as minimal as possible.